R/RLearner_classif_h2odeeplearning.R

```diff
@@ -0,0 +1,270 @@
+
+# replicate_training_data
+# Replicate the entire training dataset onto every node for faster training
+
+# single_node_mode
+# Run on a single node for fine-tuning of model parameters
+
+# shuffle_training_data
+# Enable shuffling of training data (recommended if training data is replicated and train_samples_per_iteration is close to numRows*numNodes
+
+# col_major
+# Use a column major weight matrix for input layer. Can speed up forward proagation, but might slow down backpropagation (Experimental)
+
+# average_activation
+# Average activation for sparse auto-encoder (Experimental)
+
+# sparsity_beta
+# Sparsity regularization (Experimental)
+
+# max_categorical_features
+# Max. number of categorical features, enforced via hashing Experimental)
+
+# reproducible
+# Force reproducibility on small data (will be slow - only uses 1 thread)
+
+# export_weights_and_biases
+# Whether to export Neural Network weights and biases to H2O Frames"
+
+# ...
+# extra parameters to pass onto functions (not implemented)
+
+
+# Details: https://leanpub.com/deeplearning/read
+
+#'@export
+makeRLearner.classif.h2o.deeplearning = function() {
+  makeRLearnerClassif(
+    cl = "classif.h2o.deeplearning",
+    package = "h2o",
+    par.set = makeParamSet(
+      makeLogicalLearnerParam("autoencoder", default = FALSE),
+      makeLogicalLearnerParam("use_all_factor_level", default = TRUE),
+      makeDiscreteLearnerParam("activation", values = c("Rectifier", "Tanh",
+        "TanhWithDropout", "RectifierWithDropout", "Maxout", "MaxoutWithDropout"),
+        default = "Rectifier"),
+      # FIXME: hidden can also be a list of integer vectors for grid search
+      makeIntegerVectorLearnerParam("hidden", default = c(200L, 200L),
+        len = NA_integer_, lower = 1L), 
+      makeNumericLearnerParam("epochs", default = 10L, lower = 1), # doc says can be fractional
+      makeNumericLearnerParam("train_samples_per_iteration", default = -2, lower = -2), 
+      makeIntegerLearnerParam("seed", tunable = FALSE),
+      makeLogicalLearnerParam("adaptive_rate", default = TRUE),
+      makeNumericLearnerParam("rho", default = 0.99, lower = 0), # is there a upper limit for this?
+      makeNumericLearnerParam("epsilon", default = 1e-08, lower = 1e-10, upper = 1e-4),
+      makeNumericLearnerParam("rate", default = 0.005, lower = 0, upper = 1),
+      makeNumericLearnerParam("rate_annealing", default = 1e-06, lower = 0),
+      makeNumericLearnerParam("rate_decay", default = 1, lower = 0),
+      makeNumericLearnerParam("momentum_start", default = 0),
+      makeNumericLearnerParam("momentum_ramp", default = 1e+06),
+      makeNumericLearnerParam("momentum_stable", default = 0),
+      makeLogicalLearnerParam("nesterov_accelerated_gradient", default = TRUE),
+      makeNumericLearnerParam("input_dropout_ratio", default = 0),
+      makeNumericLearnerParam("hidden_dropout_ratios", default = 0.5),
```

Comment:
Please note that `hidden_dropout_ratios` could also be a vector of the same length as the `hidden` parameter to allow a different dropout ratio in each hidden layer, see e.g. http://didericksen.github.io/deeplearning-r-h2o/.